### ISSUE TITLE
Why are normalization layers not used in the Shared Encoder?

### ISSUE THREAD

**Issue by yujieNC** [vignywang/MTLDesc#2]
Thank you for this interesting method.
Why are normalization layers not used in the Shared Encoder? Do normalization layers have adverse effects on MTLDesc?

**Comment by vignywang** [comment]
In fact, we perform L2 normalization on the extracted descriptors (see mtldesc_trainer.py), which is similar to the normalization layer.
Our approach has two benefits:
(1) The normalization layer performs channel normalization for each pixel position, while we only perform channel normalization for key point positions, so it is more efficient.
(2) The detected keypoint locations are usually sub-pixel, and the normalization layer only acts on each pixel, which will result in descriptors extracted from sub-pixel locations that are not strictly L2-normalized (i.e. The sum of squares is not 1).

In addition, we explored the effect of L2 normalization on local descriptors performance in another work, please refer to CNDesc, TMM2022 (https://ieeexplore.ieee.org/abstract/document/9761761).

**Comment by yujieNC** [comment]
Thank you for this careful reply.
I wonder why Batch Normalization is not used in the Shared Encoder.

**Comment by vignywang** [comment]
We adopted the architecture configuration of the baseline approach SuperPoint, which did not use Batch normalization.
 Perhaps there would be further performance improvements or faster convergence using BN, but we did not do such exploration in order to be consistent with the baseline. 
The personal intuition is that BN will not work because it may destroy the difference of the feature map during normalization and thus affect discrimination. 
I recommend that you verify this part through experiments, and I don't have any more information to take home.
